llap-server/src/java/org/apache/hadoop/hive/llap/io/metadata/MetadataCache.java

```diff
@@ -406,7 +405,8 @@ private boolean lockBuffer(LlapBufferOrBuffers buffers, boolean doNotifyPolicy)
     for (int i = 0; i < bufferArray.length; ++i) {
       if (lockOneBuffer(bufferArray[i], doNotifyPolicy)) continue;
       for (int j = 0; j < i; ++j) {
-        unlockSingleBuffer(buffer, true);
+        //@TODO this is a bug by it self.
```

Comment:
I would suggest to either fix the bug while you're at it or to add more documentation to this comment, why this is a bug.